### ISSUE TITLE
Возможность открытия нескольких страниц контекста 

### ISSUE THREAD

**Issue by iprst** [terratensor/regular-library#7]
…таки существует. В случае, если страница print_page открытая ранее, выгружается браузером из памяти и вкладка с ней переходит в режим сна, при вызове страницы контекста из поисковика в следующий раз откроет её в новой вкладке. 

При этом первую вкладку можно «разбудить», она запросит повторную отправку формы, при её выполнении первая страница отдаст контент, который был запрошен в ней ранее. Подозреваю, что количество страниц ограничено лишь количеством вкладок браузера и временем ожидания перед сном. 

# гиф

<details>
<summary>📽️ 📽️ 📽️ </summary>

![ezgif-2-817c39d641](https://github.com/user-attachments/assets/47cd4d78-cdd7-4eff-addb-d763b264cd3d)

</details>

**Comment by iprst** [comment]
Три разных страницы контекста тоже можно открыть. Установлено, что браузер сохраняет запрос даже после закрытия. 
 
# гиф

<details>
<summary>📽️ 📽️ 📽️</summary>

![ezgif-2-de7292e6a1](https://github.com/user-attachments/assets/ac5e0eaf-e467-411e-a6d9-dba89b982583)

</details>

**Comment by audetv** [comment]
Да, на сколько я помню, я тогда делал страницу с пост запросом (формой), чтобы у нее был всегда единый адрес, чтобы ссылкой print_page не  было смысла длиться, или ее тиражировать. Но открыть несколько контекстов можно, для запрета открытия нескольких страниц контекста я не делал действий. При необходимости надо специально углубляться в этот вопрос, для защиты, но точно придется читать сессию пользователя и задавать ему ограничения на открытие, хранить состояние, писать специальный алгоритм, а возможно сверять дополнительно и по ip адресу, и цифровому слепку устройства (fingerprint), чтобы ограничить кол-во запросов и по времени. Это все надо специально реализовывать, но есть ли в этом необходимость? То, что браузеры запоминают и могут как бы сон увести вкладку , а потом могут ее восстанавливать, да это так, но если эту открытую страницу, например обновить через f5 или кнопкой обновить страницу, то уже не будет возможности увидеть страницу контекста. И есть некоторая защита от скана ботами, чтобы не могли напрямую обращаться к урл, на который отправляется форма, и перебирать uuid (то еще занятие, если так то, но все же не невозможное). Т.е. некоторая защита от парсинга контекста. Не буду сильно углубляться, чтобы не раскрывать уязвимости публично. Да, и как показывает время, не сильно пока это кому интересно, текущей защиты возможно достаточно. (То же самое работает и на ВИБ). А 150к библиотеку используем только локально, возможно, что там вообще надо убрать эту возню с контекстом. И разрешить читать, столько сколько надо и сохранять в  закладки нужные фрагменты?

**Comment by iprst** [comment]
> есть ли в этом необходимость? То, что браузеры запоминают и могут как бы сон увести вкладку , а потом могут ее восстанавливать, да это так …есть некоторая защита от скана ботами, чтобы не могли напрямую обращаться к урл, на который отправляется форма, и перебирать uuid (то еще занятие, если так то, но все же не невозможное). Т.е. некоторая защита от парсинга контекста.

Нет, если такое поведение не является багом, то проблемы нет, текущей схемы достаточно. 

> А 150к библиотеку используем только локально, возможно, что там вообще надо убрать эту возню с контекстом. И разрешить читать, столько сколько надо и сохранять в закладки нужные фрагменты?

Надо подготовить какой-то сводный отчёт о проведённых наблюдениях за работой библиотеки и написать список полезных изменений.
